addons/payment_odoo/models/payment_token.py

```diff
@@ -0,0 +1,9 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class PaymentToken(models.Model):
+    _inherit = 'payment.token'
+
+    odoo_payment_method_type = fields.Char(string="Payment Method Type")
```

Comment:
It's the nth module defining a "method_type" field, can't we use a generic one in payment instead ?

Reply:
Some day, yes but definitely not now. Having access to this information is the exception, it's not common to many acquirers. In fact, Adyen is the only acquirer (I believe) for which we have the information *after* the payment __and__ on the token.